packages/jupyter/src/execute/leaf.tsx

```diff
@@ -119,22 +121,30 @@ export function SessionStarter({
   useEffect(() => {
     if (!core || !server || scope.session || lock.current) return;
     lock.current = true;
-    console.debug(`Jupyter: Starting session for ${pageSlug}-${notebookSlug}`);
+    console.debug(`Jupyter: Starting session for ${pageSlug}-${notebookSlug} at ${location}`);
 
     server.listRunningSessions().then((sessions) => {
       console.debug('Jupyter: running sessions', sessions);
-      const path = `/${pageSlug}-${notebookSlug}.ipynb`;
+
+      // the location gives us the correct path for the notebook including it's filename
+      // we need to replace the filename with one based on the page slug and notebook slug
+      // in order to allow for multiple independent sessions of the same notebook
+      let path = `/${pageSlug}-${notebookSlug}.ipynb`;
+      const match = location.match(/(.*)\/.*.ipynb$/);
+      if (match) {
+        path = `${match[1]}/${pageSlug}-${notebookSlug}.ipynb`;
```

Comment:
Is this line pretty much the meat of the PR?

Reply:
yep! I think path could be `${match[1]}` but convention for jupyterlab is for it to include the filename, so thebe sticks with that.